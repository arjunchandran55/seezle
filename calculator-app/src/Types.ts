const TYPES = {
    CalculatorStore: Symbol.for("CalculatorStore")
};

export { TYPES };