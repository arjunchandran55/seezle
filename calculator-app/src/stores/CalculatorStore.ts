import { injectable } from "inversify";
import { observable, action, makeObservable } from "mobx";
import {config} from "../config/config";

@injectable()
export class CalculatorStore {
    @observable inputExpression: string;
    @observable result: string;
    private defaultExpression = "0";

    constructor() {
        makeObservable(this);
        this.inputExpression = this.defaultExpression;
        this.result = "";
    }

    @action calculate = () => {
        fetch(`${config.apiUrl}/basic-calculator?expression=${(encodeURIComponent(this.inputExpression))}`)
        .then(response => response.json())
        .then(response => {
            this.result = response.expressionResult as string;
        });
    }

    @action setExpression = (expression: string) => {
            this.inputExpression = expression;
    }

    @action clearExpression = () => {
        this.inputExpression = this.defaultExpression;
        this.result = "";
    }

    @action deleteLast = () => {
        this.inputExpression = this.inputExpression.slice(0, this.inputExpression.length - 1);
        this.result = "";
    }
}