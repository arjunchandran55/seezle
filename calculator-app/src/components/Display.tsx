import { useInjection } from "inversify-react";
import { Card, Form, FormLabel } from "react-bootstrap";
import { CalculatorStore } from "../stores/CalculatorStore";
import { observer } from 'mobx-react';
import { TYPES } from "../Types";

const Display = observer(() => {
    const store: CalculatorStore = useInjection(TYPES.CalculatorStore);
    return <Card className="text-right" style={{ height: '10rem'}}>
    <Card.Body className="card bg-light">
        <Form.Control plaintext readOnly className="display lg text-right" placeholder="0" value={store.inputExpression} />
        <hr/>
        <FormLabel className="text-right text-color-red"><b className="text-right">{store.result}</b></FormLabel>
    </Card.Body>
  </Card>
});

export default Display;