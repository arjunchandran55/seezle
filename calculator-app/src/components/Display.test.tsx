import { configure, render, shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import {CalculatorStore} from "../stores/CalculatorStore";
import { Provider } from 'inversify-react';
import { Container } from 'inversify';
import { TYPES } from '../Types';
import "reflect-metadata";
import Display from "./Display";

configure({ adapter: new Adapter() }); // configure enzyme adapter

var containerMock: Container;
var storeMock: CalculatorStore;
beforeEach(() => {
  
  containerMock = new Container();
  
  storeMock = jest.genMockFromModule(
    "../stores/CalculatorStore",
  ) as CalculatorStore;

  storeMock.setExpression = jest.fn(e => { storeMock.inputExpression = e } );
  containerMock.bind(TYPES.CalculatorStore).toConstantValue(storeMock);
});

const renderComponent = (renderer: any) =>
  renderer(
    <Provider container={containerMock}>
      <Display />
    </Provider>
  );

test("Render display for calculator", () => {
  storeMock.setExpression("1+1-2");
  const comp = renderComponent(shallow);
  expect(comp).toMatchSnapshot();
});