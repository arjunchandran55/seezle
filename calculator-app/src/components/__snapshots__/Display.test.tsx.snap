// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render display for calculator 1`] = `ShallowWrapper {}`;
